Delay checking for TREUNAS_INSTALL label while installing

sonicaj opened the pull request:
This commit introduces changes where we delay checking for TRUENAS_INSTALL label while installing because USB drives can take some time to show up properly in the OS. If this still fails for some reason, we drop to shell then.

kmoore134 commented on truenas/zfs-initrd:
```diff
@@ -74,29 +74,44 @@ mkdir -p "${1}/sbin"
+	if [ -e /dev/disk/by-label/TRUENAS_INSTALL ]; then
+		break
+	else
+		echo "Unable to find TRUENAS_INSALL, sleeping for 5 seconds"
```
Typo here on TRUENAS_INSTALL.

Maybe just say "Waiting 5 seconds for root device to settle"?

kmoore134 commented on truenas/zfs-initrd:
```diff
@@ -74,29 +74,44 @@ mkdir -p "${1}/sbin"
+	fi
+done
+
+if [ -e /dev/disk/by-label/TRUENAS_INSTALL ]; then
```
No need to put this whole section into a big if/else statement. I'd do it this way:

```
if [ ! -e /dev/disk-by-label/TRUENAS_INSTALL ]  ;then
   echo "Unable to find /dev/disk/by-label/TRUENAS_INSTALL, dropping to shell"
   /bin/sh
  exit 1
fi
```

Then do the rest of the normal mount below that. Makes it look cleaner.

sonicaj committed:
Delay checking for TREUNAS_INSTALL label while installing

This commit introduces changes where we delay checking for TRUENAS_INSTALL label while installing because USB drives can take some time to show up properly in the OS. If this still fails for some reason, we drop to shell then.
